cluster_migration_core/cluster_migration_core/robot_actions/result_reporter.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import List
+
+from robot.api import ExecutionResult, ResultVisitor
+from robot.result.model import TestCase
+from robot.result import Result
+
+from cluster_migration_core.robot_actions.action_executor import ActionExecutor
+
+
+@dataclass
+class ResultReport:
+    passed: int
+    failed: int
+    skipped: int
+    not_run: int
+
+    @property
+    def result_count(self):
+        return sum([self.passed, self.failed, self.skipped, self.not_run])
+
+    def __eq__(self, other):
+        passed_same = self.passed == other.passed
+        failed_same = self.failed == other.failed
+        not_run_same = self.not_run == other.not_run
+        skipped_same = self.skipped == other.skipped
+
+        return passed_same and failed_same and not_run_same and skipped_same
+
+
+class ResultReporter:
+    """
+    Class to encapsulate the result from the provided ActionExecutors and answer questions about what they were.
+
+    The Robot Framework documentation for how to interact with result programmatically is fairly spotty, so it's quite
+    likely there's better ways to use it, but I wasn't able to figure them out.
+    """
+    def __init__(self, executors: List[ActionExecutor]):
+        output_paths = [str(e.output_xml_path.absolute()) for e in executors]
+        self._result: Result = ExecutionResult(*output_paths)
+
+    def get_result_for_tag(self, tag: str):
+        """
+        Find the result aggregated by tests for a single, specific tag (*not* a combo of tags conjoined w/ 'AND')
+        """
+
+        class GatherTestInfo(ResultVisitor):
```

Comment:
Might be nice to have a comment here as to what this class is used for, as it took me a little bit to understand how Robot framework was using this.

Reply:
Overall I think I like this more than I dislike it currently. We remove all the xml parsing logic and can easily check things like if a test passed or not, how long it took, a potential error message, etc. The downside seems to be the documentation isn't very intuitive or just not there

Reply:
Great point; will do.  This is a sophisticated API with primitive documentation.

Reply:
Added some more/better comments w/ doc links

Reply:
Much clearer thanks!